src/basic/env-util.c

```diff
@@ -16,22 +16,26 @@
 #include "strv.h"
 #include "utf8.h"
 
-#define VALID_CHARS_ENV_NAME                    \
+/* We follow bash for the character set. Different shells have different rules. */
+#define VALID_BASH_ENV_NAME_CHARS               \
         DIGITS LETTERS                          \
         "_"
 
-static bool env_name_is_valid_n(const char *e, size_t n) {
-        const char *p;
+static bool printable_portable_character(char c) {
```

Comment:
the function is a misnomer... the portable charset is larger...

Reply:
Suggestions? `nonwhitespace_portable_character`?